js/scripts.js

```diff
@@ -122,6 +136,46 @@ let pokemonRepository = (function () {
         });
     }
 
+    // Adds the search by name functionality for the Search Bar.
+    let searchBar = document.querySelector('#search-bar');
+    searchBar.addEventListener('input', function (event) {
+        let pokemonList = document.querySelectorAll('.list-group-item');
+        let filterUpperCase = searchBar.value.toUpperCase();
+
+        pokemonList.forEach(function (pokemon) {
+            if (pokemon.innerText.toUpperCase().indexOf(filterUpperCase) > -1) {
+                pokemon.style.display = '';
+            } else {
+                pokemon.style.display = 'none';
+            }
+        });
+    });
```

Comment:
This is nicely done. How about we display `No Pokemon Found from Your Search` when the search is empty? WDYT

Reply:
Thank you! Good idea, just now added the revised version.